insns/generate.py

```diff
@@ -1304,6 +1563,15 @@ def isa_propagate(suffix):
 isa_propagate("m")
 isa_propagate("mc")
 
+## B extension is Zba, Zbb, and Zbs
+for ext in ["Zba", "Zbb", "Zbs"]:
+    if "rv32i"+ext not in isa_database:
+        continue
+    isa_propagate(ext)
+    isa_propagate_pair("rv32i"+ext, "rv32ib")
+    isa_propagate_pair("rv64i"+ext, "rv64ib")
+isa_propagate_pair("rv32ib", "rv64ib")
+
```

Comment:
It would be nice to generate all variants of extensions here, you can do something like:
```python
isa_suffixes = [
    ("m", []),
    ("c", []),
    ("b", ["Zba", "Zbb", "Zbs"])
]

for suffix, sub_suffixes in isa_suffixes:
    for sub_suffix in sub_suffixes:
        isa_propagate_pair("rv32i" + sub_suffix, "rv32i" + suffix)
        isa_propagate_pair("rv64i" + sub_suffix, "rv64i" + suffix)

def propagate_isas(parts=isa_suffixes, suffix=""):
    if not parts:
        isa_propagate(suffix)
    else:
        propagate_isas(parts[1:], suffix)
        propagate_isas(parts[1:], suffix + parts[0][0])
        for part in parts[0][1]:
            propagate_isas(parts[1:], suffix + part)

propagate_isas()
```

Reply:
Yeah I was thinking about this too.  I think for a generic "any combination of extensions" it would be better to generate them during the core testing setup, rather than producing a file for each and every combination.

Reply:
I guess the advantage of having each variation is that it makes it easier to use them in a testbench outside of the genchecks.py setup, like https://github.com/YosysHQ/riscv-formal/blob/7793823fe3c281205093b1c5ac7a3aa772f5e223/cores/picorv32/complete.sby